extensions/wikibase/src/org/openrefine/wikibase/qa/EditInspector.java

```diff
@@ -206,6 +211,50 @@ public void inspect(List<EntityEdit> editBatch, WikibaseSchema schema) throws Ex
             QAWarning warning = new QAWarning("no-issue-detected", null, QAWarning.Severity.INFO, 0);
             warning.setFacetable(false);
             warningStore.addWarning(warning);
+        } else {
+            resolveWarningPropertyLabels();
+        }
+    }
+
+    private void resolveWarningPropertyLabels() throws ExecutionException {
+        if (entityCache != null) {
+            List<EntityIdValue> propertyIdValues = warningStore.getWarnings().stream()
+                    .flatMap(warning -> warning.getProperties().entrySet().stream()
+                            .filter(entry -> entry.getValue() instanceof PropertyIdValue)
+                            .map(entry -> (PropertyIdValue) entry.getValue()))
+                    .collect(Collectors.toList());
+
+            entityCache.getMultipleDocuments(propertyIdValues);
+
+            warningStore.getWarnings().stream()
+                    .forEach(warning -> {
+                        warning.getProperties().forEach((key, value) -> {
+                            if (value instanceof PropertyIdValue) {
+                                PropertyIdValue property = (PropertyIdValue) value;
+                                String label = "";
+
+                                PropertyDocument propertyDocument = (PropertyDocument) entityCache.get(property);
+                                if (propertyDocument != null && propertyDocument.getLabels() != null) {
+                                    label = propertyDocument.getLabels().get(Locale.getDefault().getLanguage()) != null
+                                            ? propertyDocument.getLabels().get(Locale.getDefault().getLanguage()).getText()
+                                            : "";
+                                }
+                                warning.setProperty(key, new SuggestedPropertyIdValue(property.getId(), property.getSiteIri(), label));
+                            } else if (value instanceof ItemIdValue) {
+                                ItemIdValue itemIdValue = (ItemIdValue) value;
+                                String itemLabel = "";
+                                ItemDocument itemDocument = (ItemDocument) entityCache.get(itemIdValue);
+
+                                if (itemDocument != null && itemDocument.getLabels() != null) {
+                                    itemLabel = itemDocument.getLabels().get(Locale.getDefault().getLanguage()) != null
+                                            ? itemDocument.getLabels().get(Locale.getDefault().getLanguage()).getText()
+                                            : "";
+                                }
+                                warning.setProperty(key,
+                                        new SuggestedPropertyIdValue(itemIdValue.getId(), itemIdValue.getSiteIri(), itemLabel));
```

Comment:
Very nice, we're getting there! To reduce the amount of duplicated code here, you could try to exploit the fact that both `ItemDocument` and `PropertyDocument` are specializations of `LabeledDocument`.
So you could merge the two `if` blocks by having a condition `if (value instanceof ItemIdValue || value instanceof PropertyIdValue)`, and then using `LabeledDocument` in place of `ItemDocument` and `PropertyDocument`. You'll still need to make a distinction at the end to create either a `SuggestedPropertyIdValue` or an `ItemIdValue` but I think the deduplication should still be worth it.

Reply:
Updated code ; duplication removed. Please review. @wetneb